storage/Devices/MdImpl.cc

```diff
@@ -915,6 +919,14 @@ namespace storage
 		break;
 	}
 
+	// adjust block size
+	if (block_size && block_size != get_region().get_block_size())
+	{
+	    Region region = Region(get_region());
+	    region.adjust_block_size(block_size);
+	    set_region(region);
+	}
```

Comment:
We would need to document the reasons for this.

I mean, I understand what the code implies - when creating a RAID, such RAID will have the bigger block size of all its underlying devices. But I miss a link to some documentation about it, maybe from `mdadm` or whatever.

Some place where I can get reply to questions such as:

 - Can I actually combine devices with different block sizes for all levels of RAID (mirroring, stripping, combinations)?
 - The biggest block size will always be used for the RAID in all cases (for all levels)?
 - What happens if some of the devices has a block size that is not divisible by the biggest one? Will mdadm fail? Will mdadm accept that but the result would be unreliable?

Reply:
(1) there seems to be no problem; I've experimented a bit
(2) it makes sense but I doubt it's documented somewhere; we'd have to check the kernel sources
(3) that's impossible as block sizes are always powers of 2

Reply:
> (3) that's impossible as block sizes are always powers of 2

Right. I didn't expressed it correctly, but I was actually refferring to the fringe case you described in the other comment. What happens in that case from the mdadm POV? Will it fail, will it create an unreliable device? My expectation was actually a device that works but is one block smaller.

Reply:
See 'Notes' above